go/vt/vitessdriver/driver.go

```diff
@@ -76,6 +78,11 @@ func OpenWithConfiguration(c Configuration) (*sql.DB, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	if len(c.GRPCDialOptions) != 0 {
+		vtgateconn.RegisterDialer(c.Protocol, grpcvtgateconn.DialWithOpts(context.TODO(), c.GRPCDialOptions...))
```

Comment:
I don't remember where protocol defaults to grpc, but do you need to do a check and set that here if protocol is blank?

Reply:
I thought that was happening in the toJSON function but it's not a pointer receiver so it's only getting propagated to callees. I'm thinking we should change the toJSON function to a pointer receiver and that will work as intended. We also won't be affecting the callers config because they are only passing a value so we have our own version at that point.

Reply:
But if they don't set a protocol and do set options, I think the way it is written will ignore those options

Reply:
I believe the `toJSON` function is only used for display purposes, and it should not be assuming that the default protocol is grpc. I'd recommend deleting that line and the `setDefaults` function.

The default is actually in the flag: `vtgateconn.VtgateProtocol`. Since the variable is exported, this can be overridden by someone who wants to change the default.